lib/vdsm/tool/configurators/multipath.py

```diff
@@ -109,3 +106,24 @@ def isconfigured():
 
     sys.stdout.write("multipath requires configuration\n")
     return NO
+
+
+def isconfigured():
+    """
+    Check the multipath daemon configuration.
+    """
+    conf_metadata_ok = _check_mpath_metadata()
+    try:
+        mpathconf.check_local_config()
+    except exception.InvalidMPathConfiguration as e:
+        # Just warn for now, do not fail the isconfigured check here.
+        sys.stdout.write(f"WARNING: {e}:\n")
```

Comment:
Using the error here means the error `__str__` cannot include all the info about the error, which is bad if we want to log it.

In the context of the tool, you should use your own message instead of using the error string representation.

Looks like you make the work harder by raising an error from mpathconf module, instead of reporting the status and using it here.

if we use:

    result = mpathconf.check_local_config()

And result is a named tuple with:
- error
- issues

We should format here the error and the issues, and we don't need to introduce a new exception. This is how lvmfilter and lvmdevices module work.

Reply:
That is also how this was coded before, until I changed it following a comment. I don't mind removing the exception and bringing that back again.